File: testdata/bench/c/BenchmarkTest00003.c
#include <string.h>
#include <sqlite3.h>

void search(const char *term) {
    char query[1024];
    sqlite3 *db;
    sqlite3_open("app.db", &db);
    strcpy(query, "SELECT * FROM products WHERE name = '");
    strcat(query, term);
    strcat(query, "'");
    sqlite3_exec(db, query, NULL, NULL, NULL);
    sqlite3_close(db);
}

File: testdata/bench/rust/BenchmarkTest00003.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Vulnerable: SQL injection via string concatenation
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("search").unwrap_or(&String::new()).clone();
    let sql = String::from("SELECT * FROM products WHERE name LIKE '%") + &param + "%'";
    // sqlx::query(&sql).execute(&pool).await
    HttpResponse::Ok().body(sql)
}
